admin_guide/http_proxies.adoc

```diff
@@ -0,0 +1,120 @@
+= Working with HTTP Proxies
+{product-author}
+{product-version}
+:icons: font
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+:description: This should be a clear and concise description of the topic.
+:keywords: comma, separated, keywords, list
+
+toc::[]
+
+== Overview
+Production environments can deny direct access to the web and instead, have an HTTP or
+HTTPS proxy available. Configuring OpenShift builds and deployments to
+use these proxies is as simple as setting standard environment variables in
+configuration or JSON files.
+
+== Importing Image Streams
+
+. Add the `*NO_PROXY*`, `*HTTP_PROXY*`, and `*HTTPS_PROXY*` environment
+variables to the *_/etc/sysconfig/openshift-master_* file:
++
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com
+----
+====
+
+. Restart the master service:
++
+----
+# systemctl restart openshift-master
+----
+
+. If you had already imported ImageStreams before creating the proxy
+configuration, delete, then re-run the process:
++
+----
+# oc delete imagestreams -n openshift --all
+# oc create -f image-streams.json -n openshift
+----
+
+== S2I Builds
+An S2I build fetches gems from a specified location. Fork that codebase and
+create a file named *_.sti/environment_*. The contents of the file are simple
+shell variables. Most libraries will look for `*NO_PROXY*`, `*HTTP_PROXY*`, and
+`*HTTPS_PROXY*` variables and react accordingly:
+
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com
+----
+====
+
+== Setting Environment Variables in Pods
+Set the following environment variables in a DeploymentConfig to pass database
+connection information. The same can be done for configuring a Pod's proxy at
+runtime:
+
+====
+----
+{
+  "apiVersion": "v1beta1",
+  "kind": "DeploymentConfig",
+  "metadata": {
+    "name": "frontend"
+  },
+  "template": {
+    "controllerTemplate": {
+      "podTemplate": {
+        "desiredState": {
+          "manifest": {
+            "containers": [
+              {
+                "env": [
+                  {
+                    "name": "HTTP_PROXY",
+                    "value": "http://USER:PASSWORD@IPADDR:PORT"
+                  },
+----
+====
+
+== Git Repository Access
+To use GitHub behind a proxy, set an environment variable on the stiStrategy:
+
+====
+----
+{
+  "stiStrategy": {
+    ...
+    "env": [
+      {
+        "Name": "HTTP_PROXY",
+        "Value": "http://USER:PASSWORD@IPADDR:PORT"
+      }
+    ]
+  }
+}
+----
+====
+
+If the variable is set on the stiStrategy it's not necessary to use the
+*_.sti/environment_* file.
```

Comment:
@bparees Sorry, I should've mentioned that I commented it out rather than deleting it. It's handy for tracking. It's not visible in published form. What the reader will be seeing instead is a copied version of this: 

https://docs.openshift.com/enterprise/3.0/dev_guide/builds.html#using-a-proxy-for-git-cloning